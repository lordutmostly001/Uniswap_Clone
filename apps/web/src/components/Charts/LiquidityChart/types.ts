import { CustomData, CustomSeriesOptions, UTCTimestamp } from 'lightweight-charts'

export interface LiquidityBarData extends CustomData {
  time: UTCTimestamp
  tick: number
  price0: string
  price1: string
  liquidity: number
  amount0Locked: number
  amount1Locked: number
}

export interface LiquidityBarProps {
  tokenAColor: string
  tokenBColor: string
  highlightColor: string
  activeTick?: number
  activeTickProgress?: number
}

export interface LiquidityBarSeriesOptions extends CustomSeriesOptions, LiquidityBarProps {
  hoveredTick?: number
}
