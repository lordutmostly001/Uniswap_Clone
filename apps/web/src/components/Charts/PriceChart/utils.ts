import { PricePoint } from 'appGraphql/data/util'
import { PriceChartData } from 'components/Charts/PriceChart'
import { CandlestickData } from 'lightweight-charts'

/**
 * Returns the minimum and maximum values in the given array of PricePoints.
 */
export function getPriceBounds(prices: PricePoint[]): { min: number; max: number } {
  if (!prices.length) {
    return { min: 0, max: 0 }
  }

  let min = prices[0].value
  let max = prices[0].value

  for (const pricePoint of prices) {
    if (pricePoint.value < min) {
      min = pricePoint.value
    }
    if (pricePoint.value > max) {
      max = pricePoint.value
    }
  }

  return { min, max }
}

/**
 * Returns the minimum and maximum values in the given array of candlestick data.
 */
export function getCandlestickPriceBounds(data: CandlestickData[]): { min: number; max: number } {
  if (!data.length) {
    return { min: 0, max: 0 }
  }

  let min = data[0].low
  let max = data[0].high

  for (const dataPoint of data) {
    if (dataPoint.low < min) {
      min = dataPoint.low
    }
    if (dataPoint.high > max) {
      max = dataPoint.high
    }
  }

  return { min, max }
}

/**
 * Returns the currentPrice or the most recent price from the data array.
 */
export function getEffectivePrice({
  currentPrice,
  data,
}: {
  currentPrice: number | undefined
  data: PriceChartData[]
}): number {
  if (currentPrice) {
    return currentPrice
  }
  return data[data.length - 1]?.value ?? 0
}
