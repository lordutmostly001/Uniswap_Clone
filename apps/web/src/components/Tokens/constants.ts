// Breakpoints specifically for the token pages

// TODO(WEB-2968): Deprecate these in the new .info project
export const MAX_WIDTH_MEDIA_BREAKPOINT = '1200px'
