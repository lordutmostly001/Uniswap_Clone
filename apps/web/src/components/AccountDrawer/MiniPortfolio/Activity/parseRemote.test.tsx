import { act, renderHook } from '@testing-library/react'
import {
  MockMint,
  MockMoonpayPurchase,
  MockNFTApproval,
  MockNFTApprovalForAll,
  MockNFTPurchase,
  MockNFTReceive,
  MockRemoveLiquidity,
  MockSenderAddress,
  MockSpamMint,
  MockSpamSwap,
  MockSwap,
  MockSwapOrder,
  MockTokenApproval,
  MockTokenReceive,
  MockTokenSend,
  MockWrap,
  mockTokenTransferInPartsFragment,
  mockTokenTransferOutPartsFragment,
  mockTransactionDetailsPartsFragment,
} from 'components/AccountDrawer/MiniPortfolio/Activity/fixtures/activity'
import {
  offchainOrderDetailsFromGraphQLTransactionActivity,
  parseRemoteActivities,
  parseSwapAmounts,
  useTimeSince,
} from 'components/AccountDrawer/MiniPortfolio/Activity/parseRemote'
import ms from 'ms'
import { MockExpiredUniswapXOrder, MockFilledUniswapXOrder, MockOpenUniswapXOrder } from 'state/signatures/fixtures'
import { DAI } from 'uniswap/src/constants/tokens'
import { UniverseChainId } from 'uniswap/src/features/chains/types'
import { TransactionType } from 'uniswap/src/features/transactions/types/transactionDetails'
import { buildCurrencyId, currencyId } from 'uniswap/src/utils/currencyId'

const swapOrderTokenChanges = {
  TokenTransfer: [mockTokenTransferOutPartsFragment, mockTokenTransferInPartsFragment],
  NftTransfer: [],
  TokenApproval: [],
  NftApproval: [],
  NftApproveForAll: [],
}

describe('parseRemote', () => {
  beforeEach(() => {
    vi.useFakeTimers()
  })
  describe('parseRemoteActivities', () => {
    it('should not parse open UniswapX order', () => {
      const result = parseRemoteActivities([MockOpenUniswapXOrder], '', vi.fn())
      expect(result).toEqual({})
    })
    it('should parse expired UniswapX order', () => {
      const result = parseRemoteActivities([MockExpiredUniswapXOrder], '', vi.fn())
      expect(result?.['someHash']).toMatchSnapshot()
    })
    it('should parse filledUniswapX order', () => {
      const result = parseRemoteActivities([MockFilledUniswapXOrder], '', vi.fn())
      expect(result?.['someHash']).toMatchSnapshot()
    })
    it('should parse NFT approval', () => {
      const result = parseRemoteActivities([MockNFTApproval], '', vi.fn())
      expect(result?.['someHash']).toMatchSnapshot()
    })
    it('should parse NFT approval for all', () => {
      const result = parseRemoteActivities([MockNFTApprovalForAll], '', vi.fn())
      expect(result?.['someHash']).toMatchSnapshot()
    })
    it('should parse NFT Mint', () => {
      const result = parseRemoteActivities([MockMint], '', vi.fn())
      expect(result?.['someHash']).toMatchSnapshot()
    })
    it('should mark spam when tx does not come from user and contains spam', () => {
      const resultFromExternal = parseRemoteActivities([MockSpamMint], '', vi.fn())
      expect(resultFromExternal?.['someHash'].isSpam).toBeTruthy()
      const resultFromUser = parseRemoteActivities([MockSpamMint], MockSenderAddress, vi.fn())
      expect(resultFromUser?.['someHash'].isSpam).toBeFalsy()
    })
    it('should parse swap', () => {
      const result = parseRemoteActivities([MockSwap], '', vi.fn().mockReturnValue('100'))
      expect(result?.['someHash']).toMatchSnapshot()
    })
    it('should not mark swaps for spam tokens as spam', () => {
      const result = parseRemoteActivities([MockSpamSwap], '', vi.fn().mockReturnValue('100'))
      expect(result?.['someHash'].isSpam).toBeFalsy()
    })
    it('should parse nft purchase', () => {
      const result = parseRemoteActivities([MockNFTPurchase], '', vi.fn().mockReturnValue('100'))
      expect(result?.['someHash']).toMatchSnapshot()
    })
    it('should parse token approval', () => {
      const result = parseRemoteActivities([MockTokenApproval], '', vi.fn())
      expect(result?.['someHash']).toMatchSnapshot()
    })
    it('should parse send', () => {
      const result = parseRemoteActivities([MockTokenSend], '', vi.fn().mockReturnValue(100))
      expect(result?.['someHash']).toMatchSnapshot()
    })
    it('should parse receive', () => {
      const result = parseRemoteActivities([MockTokenReceive], '', vi.fn().mockReturnValue(100))
      expect(result?.['someHash']).toMatchSnapshot()
    })
    it('should parse NFT receive', () => {
      const result = parseRemoteActivities([MockNFTReceive], '', vi.fn().mockReturnValue(100))
      expect(result?.['someHash']).toMatchSnapshot()
    })
    it('should parse remove liquidity', () => {
      const result = parseRemoteActivities([MockRemoveLiquidity], '', vi.fn().mockReturnValue(100))
      expect(result?.['someHash']).toMatchSnapshot()
    })
    it('should parse moonpay purchase', () => {
      const result = parseRemoteActivities([MockMoonpayPurchase], '', vi.fn().mockReturnValue(100))
      expect(result?.['someHash']).toMatchSnapshot()
    })
    it('should parse swap order', () => {
      const result = parseRemoteActivities([MockSwapOrder], '', vi.fn().mockReturnValue(100))
      expect(result?.['someHash']).toMatchSnapshot()
    })
    it('should parse eth wrap', () => {
      const result = parseRemoteActivities([MockWrap], '', vi.fn().mockReturnValue(100))
      expect(result?.['someHash']).toMatchSnapshot()
    })
  })

  describe('useTimeSince', () => {
    beforeEach(() => {
      vi.useFakeTimers()
    })

    afterEach(() => {
      vi.useRealTimers()
    })

    it('should initialize with the correct time since', () => {
      const timestamp = Math.floor(Date.now() / 1000) - 65 // 65 seconds ago
      const { result } = renderHook(() => useTimeSince(timestamp))

      expect(result.current).toBe('1m')
    })

    it('should update time since every second', async () => {
      const timestamp = Math.floor(Date.now() / 1000) - 50 // 50 seconds ago
      const { result, rerender } = renderHook(() => useTimeSince(timestamp))

      act(() => {
        vi.advanceTimersByTime(ms('1.1s'))
      })
      rerender()

      expect(result.current).toBe('51s')
    })

    it('should stop updating after 61 seconds', () => {
      const timestamp = Math.floor(Date.now() / 1000) - 61 // 61 seconds ago
      const { result, rerender } = renderHook(() => useTimeSince(timestamp))

      act(() => {
        vi.advanceTimersByTime(ms('121.1s'))
      })
      rerender()

      // maxes out at 1m
      expect(result.current).toBe('1m')
    })
  })

  describe('parseSwapAmounts', () => {
    it('should correctly parse amounts when both sent and received tokens are present', () => {
      const result = parseSwapAmounts(swapOrderTokenChanges, vi.fn().mockReturnValue('100'))
      expect(result).toEqual({
        inputAmount: '100',
        inputAmountRaw: '100000000000000000000',
        inputCurrencyAddress: DAI.address,
        outputAmount: '100',
        outputAmountRaw: '100000000000000000000',
        outputCurrencyAddress: '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2',
        sent: mockTokenTransferOutPartsFragment,
        received: mockTokenTransferInPartsFragment,
      })
    })

    it('should return undefined when sent token is missing', () => {
      const result = parseSwapAmounts(
        {
          ...swapOrderTokenChanges,
          TokenTransfer: [mockTokenTransferOutPartsFragment],
        },
        vi.fn().mockReturnValue('100'),
      )
      expect(result).toEqual(undefined)
    })
  })

  describe('offchainOrderDetailsFromGraphQLTransactionActivity', () => {
    it('should return undefined when the activity is not a swap order', () => {
      const result = offchainOrderDetailsFromGraphQLTransactionActivity(
        { ...MockSwapOrder, details: { ...mockTransactionDetailsPartsFragment, __typename: 'TransactionDetails' } },
        {
          ...swapOrderTokenChanges,
          TokenTransfer: [],
        }, // no token changes
        vi.fn().mockReturnValue('100'),
      )
      expect(result).toEqual(undefined)
    })

    it('should return the OffchainOrderDetails', () => {
      const result = offchainOrderDetailsFromGraphQLTransactionActivity(
        { ...MockSwapOrder, details: { ...mockTransactionDetailsPartsFragment, __typename: 'TransactionDetails' } },
        swapOrderTokenChanges,
        vi.fn().mockReturnValue('100'),
      )
      expect(result).toEqual({
        chainId: UniverseChainId.Mainnet,
        status: 'filled',
        id: 'tx123',
        offerer: '0xSenderAddress',
        orderHash: '0xHashValue',
        swapInfo: {
          expectedOutputCurrencyAmountRaw: '100000000000000000000',
          inputCurrencyAmountRaw: '100000000000000000000',
          inputCurrencyId: currencyId(DAI),
          isUniswapXOrder: true,
          minimumOutputCurrencyAmountRaw: '100000000000000000000',
          outputCurrencyId: buildCurrencyId(UniverseChainId.Mainnet, '0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2'),
          settledOutputCurrencyAmountRaw: '100000000000000000000',
          tradeType: 0,
          type: TransactionType.Swap,
        },
        txHash: '0xHashValue',
        addedTime: 10000,
      })
    })
  })
})
