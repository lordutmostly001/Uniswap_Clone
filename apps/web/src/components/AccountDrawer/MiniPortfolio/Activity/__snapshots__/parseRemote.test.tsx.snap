// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parseRemote > parseRemoteActivities > should parse NFT Mint 1`] = `
{
  "chainId": 1,
  "descriptor": "1 SomeCollectionName",
  "from": "0x50EC05ADe8280758E2077fcBC08D878D4aef79C3",
  "hash": "someHash",
  "isSpam": false,
  "logos": [
    "imageUrl",
  ],
  "nonce": 12345,
  "status": "confirmed",
  "timestamp": 10000,
  "title": "Minted",
  "type": "MINT",
}
`;

exports[`parseRemote > parseRemoteActivities > should parse NFT approval 1`] = `
{
  "chainId": 1,
  "descriptor": "0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045",
  "from": "0x50EC05ADe8280758E2077fcBC08D878D4aef79C3",
  "hash": "someHash",
  "isSpam": false,
  "logos": [],
  "nonce": 12345,
  "status": "confirmed",
  "timestamp": 10000,
  "title": "Unknown Approval",
  "type": "APPROVE",
}
`;

exports[`parseRemote > parseRemoteActivities > should parse NFT approval for all 1`] = `
{
  "chainId": 1,
  "descriptor": "0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045",
  "from": "0x50EC05ADe8280758E2077fcBC08D878D4aef79C3",
  "hash": "someHash",
  "isSpam": false,
  "logos": [],
  "nonce": 12345,
  "status": "confirmed",
  "timestamp": 10000,
  "title": "Unknown Approval",
  "type": "APPROVE",
}
`;

exports[`parseRemote > parseRemoteActivities > should parse NFT receive 1`] = `
{
  "chainId": 1,
  "currencies": undefined,
  "descriptor": "1 SomeCollectionName from 0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045",
  "from": "0x50EC05ADe8280758E2077fcBC08D878D4aef79C3",
  "hash": "someHash",
  "isSpam": false,
  "logos": [
    "imageUrl",
  ],
  "nonce": 12345,
  "otherAccount": "0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045",
  "status": "confirmed",
  "timestamp": 10000,
  "title": "Received",
  "type": "RECEIVE",
}
`;

exports[`parseRemote > parseRemoteActivities > should parse eth wrap 1`] = `
{
  "chainId": 1,
  "currencies": [
    NativeCurrencyImpl {
      "chainId": 1,
      "decimals": 18,
      "isNative": true,
      "isToken": false,
      "name": "Ethereum",
      "symbol": "ETH",
    },
    Token {
      "address": "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
      "buyFeeBps": undefined,
      "chainId": 1,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": "Wrapped Ether",
      "sellFeeBps": undefined,
      "symbol": "WETH",
    },
  ],
  "descriptor": "100 ETH for 100 WETH",
  "from": "0x50EC05ADe8280758E2077fcBC08D878D4aef79C3",
  "hash": "someHash",
  "isSpam": false,
  "logos": [
    "https://token-icons.s3.amazonaws.com/eth.png",
    "https://raw.githubusercontent.com/Uniswap/assets/master/blockchains/ethereum/assets/0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2/logo.png",
  ],
  "nonce": 12345,
  "status": "confirmed",
  "timestamp": 10000,
  "title": "Wrapped",
  "type": "LEND",
}
`;

exports[`parseRemote > parseRemoteActivities > should parse expired UniswapX order 1`] = `
{
  "chainId": 1,
  "currencies": [
    Token {
      "address": "0x6B175474E89094C44Da98b954EedeAC495271d0F",
      "buyFeeBps": undefined,
      "chainId": 1,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": "DAI",
      "sellFeeBps": undefined,
      "symbol": "DAI",
    },
    Token {
      "address": "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
      "buyFeeBps": undefined,
      "chainId": 1,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": "Wrapped Ether",
      "sellFeeBps": undefined,
      "symbol": "WETH",
    },
  ],
  "descriptor": "100 DAI for 200 WETH",
  "from": "someOfferer",
  "hash": "someHash",
  "logos": [
    "someUrl",
    "someUrl",
  ],
  "offchainOrderDetails": {
    "addedTime": 10000,
    "chainId": 1,
    "encodedOrder": undefined,
    "expiry": undefined,
    "id": "someId",
    "offerer": "someOfferer",
    "orderHash": "someHash",
    "status": "expired",
    "swapInfo": {
      "expectedOutputCurrencyAmountRaw": "200000000000000000000",
      "inputCurrencyAmountRaw": "100000000000000000000",
      "inputCurrencyId": "1-0x6B175474E89094C44Da98b954EedeAC495271d0F",
      "isUniswapXOrder": true,
      "minimumOutputCurrencyAmountRaw": "200000000000000000000",
      "outputCurrencyId": "1-0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
      "settledOutputCurrencyAmountRaw": undefined,
      "tradeType": 0,
      "type": "swap",
    },
    "txHash": undefined,
    "type": undefined,
  },
  "prefixIconSrc": "data:image/svg+xml,%3csvg%20width='10'%20height='14'%20viewBox='0%200%2010%2014'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M9.97119%206.19815C9.91786%206.07749%209.79854%206.00016%209.66654%206.00016H6.66654V1.00016C6.66654%200.862156%206.58189%200.738159%206.45255%200.688826C6.32255%200.638826%206.17787%200.674818%206.0852%200.776818L0.0852016%207.44349C-0.00279838%207.54149%20-0.025439%207.68149%200.028561%207.80216C0.0818943%207.92283%200.201208%208.00016%200.333208%208.00016H3.33321V13.0002C3.33321%2013.1382%203.41786%2013.2622%203.5472%2013.3115C3.58653%2013.3262%203.62654%2013.3335%203.66654%2013.3335C3.75921%2013.3335%203.84988%2013.2948%203.91455%2013.2228L9.91455%206.55616C10.0025%206.45882%2010.0245%206.31815%209.97119%206.19815Z'%20fill='url(%23paint0_linear_1816_1801)'/%3e%3cdefs%3e%3clinearGradient%20id='paint0_linear_1816_1801'%20x1='-10.1808'%20y1='-12.0005'%20x2='10.6572'%20y2='-11.6015'%20gradientUnits='userSpaceOnUse'%3e%3cstop%20stop-color='%234673FA'/%3e%3cstop%20offset='1'%20stop-color='%239646FA'/%3e%3c/linearGradient%3e%3c/defs%3e%3c/svg%3e",
  "status": "failed",
  "timestamp": 10000,
  "title": "Swap expired",
}
`;

exports[`parseRemote > parseRemoteActivities > should parse filledUniswapX order 1`] = `
{
  "chainId": 1,
  "currencies": [
    Token {
      "address": "0x6B175474E89094C44Da98b954EedeAC495271d0F",
      "buyFeeBps": undefined,
      "chainId": 1,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": "DAI",
      "sellFeeBps": undefined,
      "symbol": "DAI",
    },
    Token {
      "address": "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
      "buyFeeBps": undefined,
      "chainId": 1,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": "Wrapped Ether",
      "sellFeeBps": undefined,
      "symbol": "WETH",
    },
  ],
  "descriptor": "100 DAI for 200 WETH",
  "from": "someOfferer",
  "hash": "someHash",
  "logos": [
    "someUrl",
    "someUrl",
  ],
  "offchainOrderDetails": {
    "addedTime": 10000,
    "chainId": 1,
    "encodedOrder": undefined,
    "expiry": undefined,
    "id": "someId",
    "offerer": "someOfferer",
    "orderHash": "someHash",
    "status": "filled",
    "swapInfo": {
      "expectedOutputCurrencyAmountRaw": "200000000000000000000",
      "inputCurrencyAmountRaw": "100000000000000000000",
      "inputCurrencyId": "1-0x6B175474E89094C44Da98b954EedeAC495271d0F",
      "isUniswapXOrder": true,
      "minimumOutputCurrencyAmountRaw": "200000000000000000000",
      "outputCurrencyId": "1-0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
      "settledOutputCurrencyAmountRaw": "200000000000000000000",
      "tradeType": 0,
      "type": "swap",
    },
    "txHash": "someHash",
    "type": undefined,
  },
  "prefixIconSrc": "data:image/svg+xml,%3csvg%20width='10'%20height='14'%20viewBox='0%200%2010%2014'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M9.97119%206.19815C9.91786%206.07749%209.79854%206.00016%209.66654%206.00016H6.66654V1.00016C6.66654%200.862156%206.58189%200.738159%206.45255%200.688826C6.32255%200.638826%206.17787%200.674818%206.0852%200.776818L0.0852016%207.44349C-0.00279838%207.54149%20-0.025439%207.68149%200.028561%207.80216C0.0818943%207.92283%200.201208%208.00016%200.333208%208.00016H3.33321V13.0002C3.33321%2013.1382%203.41786%2013.2622%203.5472%2013.3115C3.58653%2013.3262%203.62654%2013.3335%203.66654%2013.3335C3.75921%2013.3335%203.84988%2013.2948%203.91455%2013.2228L9.91455%206.55616C10.0025%206.45882%2010.0245%206.31815%209.97119%206.19815Z'%20fill='url(%23paint0_linear_1816_1801)'/%3e%3cdefs%3e%3clinearGradient%20id='paint0_linear_1816_1801'%20x1='-10.1808'%20y1='-12.0005'%20x2='10.6572'%20y2='-11.6015'%20gradientUnits='userSpaceOnUse'%3e%3cstop%20stop-color='%234673FA'/%3e%3cstop%20offset='1'%20stop-color='%239646FA'/%3e%3c/linearGradient%3e%3c/defs%3e%3c/svg%3e",
  "status": "confirmed",
  "timestamp": 10000,
  "title": "Swapped",
}
`;

exports[`parseRemote > parseRemoteActivities > should parse moonpay purchase 1`] = `
{
  "chainId": 1,
  "currencies": [
    Token {
      "address": "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
      "buyFeeBps": undefined,
      "chainId": 1,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": "Wrapped Ether",
      "sellFeeBps": undefined,
      "symbol": "WETH",
    },
  ],
  "descriptor": "100 WETH for 100",
  "from": "0x50EC05ADe8280758E2077fcBC08D878D4aef79C3",
  "hash": "someHash",
  "isSpam": false,
  "logos": [
    "data:image/svg+xml,%3csvg%20width='36'%20height='36'%20viewBox='0%200%2036%2036'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3crect%20width='36'%20height='36'%20rx='18'%20fill='%237D00FF'/%3e%3cpath%20d='M24.933%2014.14a3.07%203.07%200%200%200%200-6.14%203.07%203.07%200%200%200%200%206.14ZM15.5%2028A7.495%207.495%200%200%201%208%2020.493a7.495%207.495%200%200%201%207.5-7.506c4.149%200%207.5%203.354%207.5%207.506A7.495%207.495%200%200%201%2015.5%2028Z'%20fill='%23fff'/%3e%3c/svg%3e",
  ],
  "nonce": 12345,
  "status": "confirmed",
  "timestamp": 10000,
  "title": "Purchased",
  "type": "RECEIVE",
}
`;

exports[`parseRemote > parseRemoteActivities > should parse nft purchase 1`] = `
{
  "chainId": 1,
  "descriptor": "1 SomeCollectionName",
  "from": "0x50EC05ADe8280758E2077fcBC08D878D4aef79C3",
  "hash": "someHash",
  "isSpam": false,
  "logos": [
    "imageUrl",
  ],
  "nonce": 12345,
  "status": "confirmed",
  "timestamp": 10000,
  "title": "Bought",
  "type": "SWAP",
}
`;

exports[`parseRemote > parseRemoteActivities > should parse receive 1`] = `
{
  "chainId": 1,
  "currencies": [
    Token {
      "address": "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
      "buyFeeBps": undefined,
      "chainId": 1,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": "Wrapped Ether",
      "sellFeeBps": undefined,
      "symbol": "WETH",
    },
  ],
  "descriptor": "100 WETH from 0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045",
  "from": "0x50EC05ADe8280758E2077fcBC08D878D4aef79C3",
  "hash": "someHash",
  "isSpam": false,
  "logos": [
    "logoUrl",
  ],
  "nonce": 12345,
  "otherAccount": "0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045",
  "status": "confirmed",
  "timestamp": 10000,
  "title": "Received",
  "type": "RECEIVE",
}
`;

exports[`parseRemote > parseRemoteActivities > should parse remove liquidity 1`] = `
{
  "chainId": 1,
  "currencies": [
    Token {
      "address": "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
      "buyFeeBps": undefined,
      "chainId": 1,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": "Wrapped Ether",
      "sellFeeBps": undefined,
      "symbol": "WETH",
    },
    Token {
      "address": "0x6B175474E89094C44Da98b954EedeAC495271d0F",
      "buyFeeBps": undefined,
      "chainId": 1,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": "DAI",
      "sellFeeBps": undefined,
      "symbol": "DAI",
    },
  ],
  "descriptor": "100 WETH and 100 DAI",
  "from": "0x50EC05ADe8280758E2077fcBC08D878D4aef79C3",
  "hash": "someHash",
  "isSpam": false,
  "logos": [
    "logoUrl",
    "logoUrl",
  ],
  "nonce": 12345,
  "status": "confirmed",
  "timestamp": 10000,
  "title": "Removed liquidity",
  "type": "RECEIVE",
}
`;

exports[`parseRemote > parseRemoteActivities > should parse send 1`] = `
{
  "chainId": 1,
  "currencies": [
    Token {
      "address": "0x6B175474E89094C44Da98b954EedeAC495271d0F",
      "buyFeeBps": undefined,
      "chainId": 1,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": "DAI",
      "sellFeeBps": undefined,
      "symbol": "DAI",
    },
  ],
  "descriptor": "100 DAI to 0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045",
  "from": "0x50EC05ADe8280758E2077fcBC08D878D4aef79C3",
  "hash": "someHash",
  "isSpam": false,
  "logos": [
    "logoUrl",
  ],
  "nonce": 12345,
  "otherAccount": "0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045",
  "status": "confirmed",
  "timestamp": 10000,
  "title": "Sent",
  "type": "SEND",
}
`;

exports[`parseRemote > parseRemoteActivities > should parse swap 1`] = `
{
  "chainId": 1,
  "currencies": [
    Token {
      "address": "0x6B175474E89094C44Da98b954EedeAC495271d0F",
      "buyFeeBps": undefined,
      "chainId": 1,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": "DAI",
      "sellFeeBps": undefined,
      "symbol": "DAI",
    },
    Token {
      "address": "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
      "buyFeeBps": undefined,
      "chainId": 1,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": "Wrapped Ether",
      "sellFeeBps": undefined,
      "symbol": "WETH",
    },
  ],
  "descriptor": "100 DAI for 100 WETH",
  "from": "0x50EC05ADe8280758E2077fcBC08D878D4aef79C3",
  "hash": "someHash",
  "isSpam": false,
  "logos": [
    "logoUrl",
  ],
  "nonce": 12345,
  "status": "confirmed",
  "timestamp": 10000,
  "title": "Swapped",
  "type": "SWAP",
}
`;

exports[`parseRemote > parseRemoteActivities > should parse swap order 1`] = `
{
  "chainId": 1,
  "currencies": [
    Token {
      "address": "0x6B175474E89094C44Da98b954EedeAC495271d0F",
      "buyFeeBps": undefined,
      "chainId": 1,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": "DAI",
      "sellFeeBps": undefined,
      "symbol": "DAI",
    },
    Token {
      "address": "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
      "buyFeeBps": undefined,
      "chainId": 1,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": "Wrapped Ether",
      "sellFeeBps": undefined,
      "symbol": "WETH",
    },
  ],
  "descriptor": "100 DAI for 100 WETH",
  "from": "0x50EC05ADe8280758E2077fcBC08D878D4aef79C3",
  "hash": "someHash",
  "isSpam": false,
  "logos": [
    "logoUrl",
  ],
  "nonce": 12345,
  "offchainOrderDetails": {
    "addedTime": 10000,
    "chainId": 1,
    "id": "transactionId",
    "offerer": "0x50EC05ADe8280758E2077fcBC08D878D4aef79C3",
    "orderHash": "someHash",
    "status": "filled",
    "swapInfo": {
      "expectedOutputCurrencyAmountRaw": "100000000000000000000",
      "inputCurrencyAmountRaw": "100000000000000000000",
      "inputCurrencyId": "1-0x6B175474E89094C44Da98b954EedeAC495271d0F",
      "isUniswapXOrder": true,
      "minimumOutputCurrencyAmountRaw": "100000000000000000000",
      "outputCurrencyId": "1-0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2",
      "settledOutputCurrencyAmountRaw": "100000000000000000000",
      "tradeType": 0,
      "type": "swap",
    },
    "txHash": "someHash",
  },
  "prefixIconSrc": "data:image/svg+xml,%3csvg%20width='10'%20height='14'%20viewBox='0%200%2010%2014'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M9.97119%206.19815C9.91786%206.07749%209.79854%206.00016%209.66654%206.00016H6.66654V1.00016C6.66654%200.862156%206.58189%200.738159%206.45255%200.688826C6.32255%200.638826%206.17787%200.674818%206.0852%200.776818L0.0852016%207.44349C-0.00279838%207.54149%20-0.025439%207.68149%200.028561%207.80216C0.0818943%207.92283%200.201208%208.00016%200.333208%208.00016H3.33321V13.0002C3.33321%2013.1382%203.41786%2013.2622%203.5472%2013.3115C3.58653%2013.3262%203.62654%2013.3335%203.66654%2013.3335C3.75921%2013.3335%203.84988%2013.2948%203.91455%2013.2228L9.91455%206.55616C10.0025%206.45882%2010.0245%206.31815%209.97119%206.19815Z'%20fill='url(%23paint0_linear_1816_1801)'/%3e%3cdefs%3e%3clinearGradient%20id='paint0_linear_1816_1801'%20x1='-10.1808'%20y1='-12.0005'%20x2='10.6572'%20y2='-11.6015'%20gradientUnits='userSpaceOnUse'%3e%3cstop%20stop-color='%234673FA'/%3e%3cstop%20offset='1'%20stop-color='%239646FA'/%3e%3c/linearGradient%3e%3c/defs%3e%3c/svg%3e",
  "status": "confirmed",
  "timestamp": 10000,
  "title": "Swapped",
  "type": "SWAP_ORDER",
}
`;

exports[`parseRemote > parseRemoteActivities > should parse token approval 1`] = `
{
  "chainId": 1,
  "currencies": [
    Token {
      "address": "0x6B175474E89094C44Da98b954EedeAC495271d0F",
      "buyFeeBps": undefined,
      "chainId": 1,
      "decimals": 18,
      "isNative": false,
      "isToken": true,
      "name": "DAI",
      "sellFeeBps": undefined,
      "symbol": "DAI",
    },
  ],
  "descriptor": "DAI",
  "from": "0x50EC05ADe8280758E2077fcBC08D878D4aef79C3",
  "hash": "someHash",
  "isSpam": false,
  "logos": [
    "logoUrl",
  ],
  "nonce": 12345,
  "status": "confirmed",
  "timestamp": 10000,
  "title": "Approved",
  "type": "APPROVE",
}
`;
